🐛 Github Reporter not generating annotations correctly

### Environment information

Commands are being run in a `Warpbuild` Github Actions runner (specifically `warp-ubuntu-latest-x64-2x`), but i've tested and this happens on `ubuntu-latest` too

```block
CLI:
  Version:                      1.8.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "unknown"
  JS_RUNTIME_VERSION:           "v18.19.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/4.1.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

1. Set up biome in a project with linting errors
2. Run `yarn biome ci --reporter=github` in an actions workflow/step

### Expected result

Expected: Annotation output should turn into an annotation in the Github Actions execution logs/`Files Changed` view

Actual: The annotation is output as a string and nothing happens. i.e. I see this in the console:
```
::error title=lint/correctness/useJsxKeyInIterable,file=path/to/file.tsx,line=36,endLine=36,col=20,endColumn=48::Missing key property for this element in iterable.
Error: Process completed with exit code 1.
```
Instead of this, and all the other annotations stuff:
```
Error: Missing key property for this element in iterable.
```

I can confirm that the annotation _format_ is correct - if I manually `echo` the output it turns into an annotation correctly, and it's only from `yarn biome ci` that it fails to work.


### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,8 @@ our [guidelines for writing a good changelog entry](https://github.com/biomejs/b
 
 - Fix [#3104](https://github.com/biomejs/biome/issues/3104) by suppressing node warnings when using `biome migrate`. Contributed by @SuperchupuDev
 
+- Force colors to be off when using the GitHub reporter to properly create annotations in GitHub actions ([#3148](https://github.com/biomejs/biome/issues/3148)). Contributed by @Sec-ant
+
 ### Parser
 
 #### New features

--- a/crates/biome_cli/src/cli_options.rs
+++ b/crates/biome_cli/src/cli_options.rs
@@ -51,7 +51,7 @@ pub struct CliOptions {
     /// Allows to change how diagnostics and summary are reported.
     #[bpaf(
         long("reporter"),
-        argument("json|json-pretty|summary"),
+        argument("json|json-pretty|github|junit|summary"),
         fallback(CliReporter::default())
     )]
     pub reporter: CliReporter,

--- a/crates/biome_cli/src/commands/mod.rs
+++ b/crates/biome_cli/src/commands/mod.rs
@@ -1,5 +1,5 @@
 use crate::changed::{get_changed_files, get_staged_files};
-use crate::cli_options::{cli_options, CliOptions, ColorsArg};
+use crate::cli_options::{cli_options, CliOptions, CliReporter, ColorsArg};
 use crate::diagnostics::{DeprecatedArgument, DeprecatedConfigurationFile};
 use crate::execute::Stdin;
 use crate::logging::LoggingKind;
@@ -476,7 +476,19 @@ impl BiomeCommand {
 
     pub const fn get_color(&self) -> Option<&ColorsArg> {
         match self.cli_options() {
-            Some(cli_options) => cli_options.colors.as_ref(),
+            Some(cli_options) => {
+                // To properly display GitHub annotations we need to disable colors
+                if matches!(cli_options.reporter, CliReporter::GitHub) {
+                    return Some(&ColorsArg::Off);
+                }
+                // We want force colors in CI, to give e better UX experience
+                // Unless users explicitly set the colors flag
+                if matches!(self, BiomeCommand::Ci { .. }) && cli_options.colors.is_none() {
+                    return Some(&ColorsArg::Force);
+                }
+                // Normal behaviors
+                cli_options.colors.as_ref()
+            }
             None => None,
         }
     }

--- a/crates/biome_cli/src/main.rs
+++ b/crates/biome_cli/src/main.rs
@@ -8,7 +8,7 @@ use biome_cli::{
     biome_command, open_transport, setup_panic_handler, to_color_mode, BiomeCommand, CliDiagnostic,
     CliSession,
 };
-use biome_console::{markup, ColorMode, ConsoleExt, EnvConsole};
+use biome_console::{markup, ConsoleExt, EnvConsole};
 use biome_diagnostics::{set_bottom_frame, Diagnostic, PrintDiagnostic};
 use biome_service::workspace;
 use std::process::{ExitCode, Termination};
@@ -37,17 +37,7 @@ fn main() -> ExitCode {
     let mut console = EnvConsole::default();
     let command = biome_command().fallback_to_usage().run();
 
-    let color_mode = to_color_mode(command.get_color());
-    // we want force colours in CI, to give e better UX experience
-    if let BiomeCommand::Ci { cli_options, .. } = &command {
-        if cli_options.colors.is_none() {
-            console.set_color(ColorMode::Enabled);
-        } else {
-            console.set_color(color_mode);
-        }
-    } else {
-        console.set_color(color_mode);
-    }
+    console.set_color(to_color_mode(command.get_color()));
 
     let is_verbose = command.is_verbose();
     let result = run_workspace(&mut console, command);

--- a/packages/tailwindcss-config-analyzer/src/generate-tailwind-preset.ts
+++ b/packages/tailwindcss-config-analyzer/src/generate-tailwind-preset.ts
@@ -60,7 +60,7 @@ function generateVariants(variants: SortConfig["variants"]) {
 	for (const { name } of variants) {
 		output += `${INDENT}"${name}",\n`
 	}
-	output += `];\n`;
+	output += "];\n";
 	return output;
 }
 

